core/platform/datastore/gae_datastore_services.py

```diff
@@ -23,6 +23,109 @@
 from google.appengine.ext import ndb
 
 
+Model = ndb.Model
+Key = ndb.Key
+
+BooleanProperty = ndb.BooleanProperty
+DateTimeProperty = ndb.DateTimeProperty
+FloatProperty = ndb.FloatProperty
+IntegerProperty = ndb.IntegerProperty
+JsonProperty = ndb.JsonProperty
+StringProperty = ndb.StringProperty
+TextProperty = ndb.TextProperty
+UserProperty = ndb.UserProperty
+
+
+def put_multi(models):
+    """Stores a sequence of Model instances.
+
+    Args:
+        models: datastore_services.Model. A sequence of Model instances.
+
+    Returns:
+        list(str). A list with the stored keys.
+    """
+    return ndb.put_multi(models)
+
+
+def put_multi_async(models, update_last_updated_time=True):
+    """Stores a sequence of Model instances asynchronously.
+
+    Args:
+        models: datastore_services.Model. A sequence of Model instances.
+        update_last_updated_time: bool. Whether to update the last_updated field
+            of the entities.
+
+    Returns:
+        list(future). A list of futures.
+    """
+    return ndb.put_multi_async(
+        models, update_last_updated_time=update_last_updated_time)
+
+
+def delete_multi(keys):
+    """Deletes a sequence of keys.
+
+    Args:
+        keys: list(str). A sequence of keys.
+
+    Returns:
+        list(None). A list of Nones, one per deleted model.
+    """
+    return ndb.delete_multi(keys)
+
+
+def transaction(callback):
+    """Run a callback in a transaction.
+
+    To pass arguments to a callback function, use a lambda, for example:
+
+    def my_callback(key, inc): do_something()
+    transaction(lambda: my_callback(Key(...), 1))
+
+    Args:
+        callback: callable. A function or tasklet to be called.
+
+    Returns:
+        *. Whatever callback() returns.
+
+    Raises:
+        Exception. Whatever callback() raises, or
+            datastore_errors.TransactionFailedError when the transaction failed.
+    """
+    return ndb.transaction(
+        callback, xg=True, propagation=ndb.TransactionOptions.ALLOWED)
+
+
+def query_everything():
+    """Returns a query that targets every single entity in the datastore."""
+    return ndb.Query()
+
+
+def all_of(*nodes):
```

Comment:
Should we keep the naming to `and` and `or` so that developers are not confused? Or maybe instead of `all_of` use just `all`.

Reply:
I decided on `all_of`/`any_of` because they match the name of the builtin `all_of` and `any_of`. I'm not really sure if we should give it a different name.

@seanlip WDYT?

Reply:
I'm fine with either. I slightly lean towards all_of/any_of for readability but I don't have a particularly strong preference.

Reply:
Going to stick with current naming since voting is (slightly) in favor of it.